runtime/src/rpc.rs

```diff
@@ -83,21 +79,27 @@ impl PolkaBtcProvider {
             ext_client,
             signer: Arc::new(RwLock::new(signer)),
             account_id,
-            only_read_finalized_storage: true,
         })
     }
 
     pub async fn from_url(
         url: &String,
         signer: PairSigner<PolkaBtcRuntime, KeyPair>,
     ) -> Result<Self, Error> {
-        let rpc_client = if url.starts_with("ws://") || url.starts_with("wss://") {
-            jsonrpsee::ws_client(url).await?
-        } else {
-            jsonrpsee::http_client(url)
+        let parsed_url = url::Url::parse(url)?;
+        let path = parsed_url.path();
+        let config = WsConfig {
+            url,
+            max_request_body_size: 10 * 1024 * 1024,
+            request_timeout: None,
+            connection_timeout: Duration::from_secs(10),
+            origin: None,
+            handshake_url: path.into(),
+            max_concurrent_requests_capacity: 256,
+            max_notifs_per_subscription_capacity: 4,
```

Comment:
Probably we want this more than 4 - in our fork we had set this to 128 to prevent freezing